api/src/main/java/eu/darkbot/api/managers/QuestAPI.java

```diff
@@ -0,0 +1,194 @@
+package eu.darkbot.api.managers;
+
+import eu.darkbot.api.API;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+/**
+ * Provides access to data for the Quests in-game.
+ */
+public interface QuestAPI extends API.Singleton {
+
+    /**
+     * @return The quest that is currently active and in view.
+     *         Even if there are several accepted quests, only the selected one is
+     *         returned.
+     */
+    @Nullable Quest getCurrentQuestDisplayed();
+
+    /**
+     * @return The quest that is currently selected in the QuestGiver
+     */
+    @Nullable Quest getCurrentQuestSeleted();
+
+    /**
+     * @return Returns additional information on the selected quest
+     */
+    @Nullable QuestListItem getCurrentQuestInfoSeleted();
+
+    /**
+     * @return Returns a list of current quests, only has data when QuestGiver is
+     *         opened
+     */
+    @Nullable List<QuestListItem> getCurrestQuests();
+
+    /**
+     * @return Returns true if the quest giver is open
+     */
+    boolean isVisibleQuestGiver();
+
+    /**
+     * @return Returns the tab position that is being displayed in the QuestGiver
+     */
+    int getTabSelected();
+
+    /**
+     * Quest Object
+     */
+    interface Quest {
+
+        /**
+         * @return The Quest ID
+         */
+        int getId();
+
+        /**
+         * @return If the quest is Active
+         */
+        boolean isActive();
+
+        /**
+         * @return The quest category
+         */
+        String getCategory();
+
+        /**
+         * @return The title of the quest in the client's language
+         */
+        String getTitle();
+
+        /**
+         * @return The description of the quest in the client's language
+         */
+        String getDescription();
+
+        /**
+         * @return The number of requirements the quest has to be completed.
+         */
+        int getRequirementsCount();
+
+        /**
+         * @return True if completed
+         */
+        boolean isCompleted();
+
+        /**
+         * @return List of requirements for completion
+         */
+        List<Requirement> getRequirements();
+
+        /**
+         * @return List of rewards to be obtained by completing the quest
+         */
+        List<Reward> getRewards();
+    }
+
+    /**
+     * QuestListItem Object
+     */
+    interface QuestListItem {
+
+        /**
+         * @return The Quest ID
+         */
+        int getId();
+
+        /**
+         * @return Level required to activate it
+         */
+        int getLevelRequired();
+
+        /**
+         * @return True if selected in the interface
+         */
+        boolean isSelected();
+
+        /**
+         * @return True if completed
+         */
+        boolean isCompleted();
+
+        /**
+         * @return Quest Title
+         */
+        String getTitle();
+
+        /**
+         * @return Type of quest
+         */
+        String getType();
```

Comment:
Are these known types? can it be an enum?

Reply:
I have a few that I have seen in the tests but I don't know the whole list, nor have I seen that they are of great importance.